pkgs/top-level/all-packages.nix

```diff
@@ -3596,7 +3596,9 @@ with pkgs;
 
   reg = callPackage ../tools/virtualization/reg { };
 
-  river = callPackage ../applications/window-managers/river { };
+  river = callPackage ../applications/window-managers/river {
+    zig = zig_0_8_1;
+  };
```

Comment:
Note: river 0.1.2 introduces zig 0.9 support https://github.com/riverwm/river/releases/tag/v0.1.2

Reply:
This change was backported from https://github.com/NixOS/nixpkgs/pull/151518, I'll rebase now that it's merged.